src/commonMain/kotlin/org/cru/godtools/tool/state/State.kt

```diff
@@ -0,0 +1,23 @@
+package org.cru.godtools.tool.state
+
+import org.cru.godtools.tool.internal.parcel.Parcelable
+import org.cru.godtools.tool.internal.parcel.Parcelize
+import org.cru.godtools.tool.model.EVENT_NAMESPACE_STATE
+import org.cru.godtools.tool.model.EventId
+
+@Parcelize
+class State internal constructor(private val state: MutableMap<String, List<String>>) : Parcelable {
+    constructor() : this(mutableMapOf<String, List<String>>())
+
+    operator fun get(key: String) = state[key]?.firstOrNull()
+    fun getAll(key: String) = state[key].orEmpty()
+
+    operator fun set(key: String, value: String?) = state.set(key, listOfNotNull(value))
+    operator fun set(key: String, values: List<String>?) = state.set(key, values.orEmpty())
+    fun set(key: String, vararg values: String) = state.set(key, values.toList())
```

Comment:
`operator` functions are functions that overload common operators. https://kotlinlang.org/docs/operator-overloading.html#indexed-access-operator

so, `operator fun set(key: Any?, value: Any?)` allows us to call it using this syntax: `obj[key] = value` instead of having to call `obj.set(key, value)`

Reply:
functions can share the same name, uniqueness is enforced on the combination of name + arg types